sympy/core/numbers.py

```diff
@@ -29,6 +29,7 @@
     prec_to_dps)
 from sympy.utilities.misc import debug, filldedent
 from .evaluate import global_evaluate
+from sympy.core.compatibility import lru_cache
```

Comment:
This should be added to the other imports from compatibility above. Alternatively, `lru_cache` could be imported from `sympy.core.cache`. There it could also be redefined as `lru_cache = fastcache.clru_cache`, if `fastcache` can be imported successfully,